### PR TITLE
ssh: Enforce not connecting to unknown hosts even as secondary machines

### PR DESCRIPTION
**Pull request by petervo**

### CODE REVIEW

**Comment by petervo** [comment]
@dperpeet, this *may* be worth a backport. If not both commits at least the first one.

File: pkg/lib/machine-dialogs.js [hunk 1]
```diff
@@ -74,7 +75,8 @@
         "invalid-hostkey" : translate_and_init(invalid_hostkey_tmpl),
         "not-supported" : translate_and_init(not_supported_tmpl),
         "sync-users" : translate_and_init(sync_users_tmpl),
-        "unknown-hostkey" : translate_and_init(unknown_hosts_tmpl)
+        "unknown-hostkey" : translate_and_init(unknown_hosts_tmpl),
+        "unknown-host" : translate_and_init(unknown_hosts_tmpl)
```
Comment (martinpitt) [inline pkg/lib/machine-dialogs.js]: OOI, what is the semantic difference between these two? Is "unknown-hostkey" just for the case that the destination name/IP is already known, but the key changed? Because the latter would be an entirely different "class" of error which should never be overridable in our UI, but we should obviously handle an entirely unknown host. I thought that "changed key" case was "invalid-hostkey".

The code below treats these two pretty much equivalently, so I wonder if the two error codes could just be unified?

File: pkg/lib/machine-dialogs.js [hunk 2]
```diff
@@ -147,6 +149,7 @@
                 conn_options['temp-session'] = false; /* Compatiblity option */
                 conn_options['session'] = 'shared';
                 conn_options['host-key'] = machine.host_key;
+                conn_options['unknown-host'] = "1";
```
Comment (martinpitt) [inline pkg/lib/machine-dialogs.js]: Why is this necessary? Apparently we do have a host key? (is `!on_disk` is being used for machines other than the local host?)

File: pkg/lib/machine-dialogs.js (hunk 1 above)
Comment (stefwalter) [inline pkg/lib/machine-dialogs.js]: "invalid-hostkey" we knew a key for a host, but it is no longer the same. It is handled very differently from "unknown-hostkey": The user is not allowed to click through or resolve the former through the UI.

"unknown-host" is a host that Cockpit doesn't know about (yet). We can't just go connecting to random hosts without the user or code adding that host.

Comment (martinpitt) [inline pkg/lib/machine-dialogs.js]: Of course not --  but how is "unknown-host" different from "unknown-hostkey"?

Comment (petervo) [inline pkg/lib/machine-dialogs.js]: The difference is that if you get unknown-hostkey you should expect there to be a host-key field in the response json. We know that because we had permission to talk to the server and get it. With unknown-host we don't know the host-key because we don't have permission to talk to the server. To get that permission it needs to be a private channel or have the unknown-host param.

File: pkg/lib/machine-dialogs.js (hunk 2 above)
Comment (petervo) [inline pkg/lib/machine-dialogs.js]: Yes it is used for not localhost machines. We have a host-key but we need to preserve the fact we have permission to talk to this host. Normally that permission is only for private sessions. But in these cases it needs to be a public session but we can't write the key to disk. So the presence of this key signifies that it's ok to talk to the otherwise unknown host. Because we can't write the key to disk,

Does it need a rename to make it clearer?

Comment (martinpitt) [inline pkg/lib/machine-dialogs.js]: Ah, and the `COCKPIT_SSH_ALLOW_UNKNOWN=${unknown-host}` bridge match is the vehicle in this case. Thanks for the explanation!
